tests: gnrc_ipv6_nib_6ln fails

<!--
If your issue is a usage question, please submit it to the user mailing-list
users@riot-os.org or to the developer mailing-list devel@riot-os.org.
If your issue is related to security, please submit it to the security
mailing-list security@riot-os.org.
-->

#### Description

Test application `tests/gnrc_ipv6_nib_6ln` fails with 4 errors, on macOS *and* Linux.

#### Steps to reproduce the issue

run `make tests` for named tests app

#### Expected results

success without error

#### Actual results

4 errors:

```
/RIOT/tests/gnrc_ipv6_nib_6ln/bin/native/tests_gnrc_ipv6_nib_6ln.elf   
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2018.04-devel-90-gb138a-mobi25.inet.haw-hamburg.de-fix_tests_180129)
.........................................
tests.test_handle_pkt__rtr_adv__success_pio_L0 (tests/gnrc_ipv6_nib_6ln/main.c 1075) _PIO_PFX_LTIME < prefix.valid_until
..
tests.test_handle_pkt__rtr_adv__success_pio_LA (tests/gnrc_ipv6_nib_6ln/main.c 1075) _PIO_PFX_LTIME < prefix.valid_until
...
tests.test_handle_pkt__rtr_adv__success_6co_sl2ao_mtuo_pio_L0 (tests/gnrc_ipv6_nib_6ln/main.c 1075) _PIO_PFX_LTIME < prefix.valid_until
..
tests.test_handle_pkt__rtr_adv__success_6co_sl2ao_mtuo_pio_LA (tests/gnrc_ipv6_nib_6ln/main.c 1075) _PIO_PFX_LTIME < prefix.valid_until
.
run 49 failures 4
Timeout in expect script at "child.expect(r"OK \(\d+ tests\)")" (tests/gnrc_ipv6_nib_6ln/tests/01-run.py:15)

make: *** [test] Error 1
```

#### Versions

OS: macOS 10.13.03 with

```
Installed toolchain versions
----------------------------
          native gcc: gcc (Homebrew GCC 7.2.0_1) 7.2.0
          msp430-gcc: missing/error
             avr-gcc: avr-gcc (GCC) 7.3.0
   arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
    mips-mti-elf-gcc: missing/error
               clang: Apple LLVM version 9.0.0 (clang-900.0.39.2)
arm-none-eabi-newlib: "2.5.0"
 mips-mti-elf-newlib: missing/error
            avr-libc: "2.0.0" ("20150208")
            cppcheck: Cppcheck 1.82
          coccinelle: spatch version 1.0.6 compiled with OCaml version 4.05.0
                 git: git version 2.16.1
```

Linux: CentOS 7 with

```
Installed toolchain versions
----------------------------
          native gcc: gcc (GCC) 6.3.1 20170216 (Red Hat 6.3.1-3)
          msp430-gcc: missing/error
             avr-gcc: missing/error
   arm-none-eabi-gcc: missing/error
    mips-mti-elf-gcc: missing/error
               clang: clang version 3.4.2 (tags/RELEASE_34/dot2-final)
arm-none-eabi-newlib: missing/error
 mips-mti-elf-newlib: missing/error
            avr-libc: missing/error (missing/error)
            cppcheck: missing
          coccinelle: missing
                 git: git version 1.8.3.1
```

--- a/sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c
+++ b/sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c
@@ -764,12 +764,17 @@ _nib_offl_entry_t *_nib_pl_add(unsigned iface,
         if (pref_ltime != UINT32_MAX) {
             _evtimer_add(dst, GNRC_IPV6_NIB_PFX_TIMEOUT, &dst->pfx_timeout,
                          pref_ltime);
-            if (((pref_ltime + now) == UINT32_MAX) && (now != 0)) {
-                pref_ltime++;
+            /* ignore capped of preferred lifetimes from sec to ms conversion */
+            if (pref_ltime < (UINT32_MAX - 1)) {
+                /* prevent pref_ltime from becoming UINT32_MAX */
+                if (((pref_ltime + now) == UINT32_MAX) && (now != 0)) {
+                    pref_ltime++;
+                }
+                pref_ltime += now;
             }
-            pref_ltime += now;
         }
-        if (valid_ltime != UINT32_MAX) {
+        /* ignore capped of valid lifetimes from sec to ms conversion */
+        if (valid_ltime < (UINT32_MAX - 1)) {
             /* prevent valid_ltime from becoming UINT32_MAX */
             if ((valid_ltime + now) == UINT32_MAX) {
                 valid_ltime++;

--- a/tests/gnrc_ipv6_nib/main.c
+++ b/tests/gnrc_ipv6_nib/main.c
@@ -1114,8 +1114,8 @@ static void test_handle_pkt__rtr_adv__success(uint8_t rtr_adv_flags,
                                 "Unexpected prefix configured");
             TEST_ASSERT_EQUAL_INT(_LOC_GB_PFX_LEN, prefix.pfx_len);
             TEST_ASSERT_EQUAL_INT(_mock_netif->pid, prefix.iface);
-            TEST_ASSERT(_PIO_PFX_LTIME < prefix.valid_until);
-            TEST_ASSERT(_PIO_PFX_LTIME < prefix.pref_until);
+            TEST_ASSERT((_PIO_PFX_LTIME / MS_PER_SEC) < prefix.valid_until);
+            TEST_ASSERT((_PIO_PFX_LTIME / MS_PER_SEC) < prefix.pref_until);
         }
     }
     if (!pio || !(pio_flags & NDP_OPT_PI_FLAGS_L)) {

--- a/tests/gnrc_ipv6_nib_6ln/main.c
+++ b/tests/gnrc_ipv6_nib_6ln/main.c
@@ -1072,8 +1072,8 @@ static void test_handle_pkt__rtr_adv__success(uint8_t rtr_adv_flags,
                                 "Unexpected prefix configured");
             TEST_ASSERT_EQUAL_INT(_LOC_GB_PFX_LEN, prefix.pfx_len);
             TEST_ASSERT_EQUAL_INT(_mock_netif->pid, prefix.iface);
-            TEST_ASSERT(_PIO_PFX_LTIME < prefix.valid_until);
-            TEST_ASSERT(_PIO_PFX_LTIME < prefix.pref_until);
+            TEST_ASSERT((_PIO_PFX_LTIME / MS_PER_SEC) < prefix.valid_until);
+            TEST_ASSERT((_PIO_PFX_LTIME / MS_PER_SEC) < prefix.pref_until);
         }
     }
     if (!pio) {

